src/util.cpp

```diff
@@ -32,6 +33,66 @@
 
 #include <KParts/ReadOnlyPart>
 
+namespace {
+struct ElidedText
+{
+    QString first;
+    QString second;
+};
+
+ElidedText elideArguments(const QString symbolText)
+{
+    QChar quoteChar = QLatin1Char('"');
+    int numQuotes = 0;
+    int depth = 0;
+
+    int endEliding = symbolText.length();
+    int startEliding = 0;
+
+    for (int i = symbolText.length() - 1; i > 0; i--) {
```

Comment:
please add some basic comments that document what this code is doing on a high level